core/capabilities/ccip/ccipsolana/executecodec.go

```diff
@@ -211,47 +196,53 @@ func parseExtraArgsMap(input map[string]any) (ccip_router.SVMExtraArgs, error) {
 	var out ccip_router.SVMExtraArgs
 
 	// Iterate through the expected fields in the struct
+	// the field name should match with the one in SVMExtraArgsV1
+	// https://github.com/smartcontractkit/chainlink/blob/33c0bda696b0ed97f587a46eacd5c65bed9fb2c1/contracts/src/v0.8/ccip/libraries/Client.sol#L57
 	for fieldName, fieldValue := range input {
-		switch fieldName {
-		case "ComputeUnits":
+		lowercase := strings.ToLower(fieldName)
+		switch lowercase {
+		case "computeunits":
 			// Expect uint32
 			if v, ok := fieldValue.(uint32); ok {
 				out.ComputeUnits = v
 			} else {
 				return out, errors.New("invalid type for ComputeUnits, expected uint32")
 			}
-		case "IsWritableBitmap":
+		case "accountiswritablebitmap":
 			// Expect uint64
 			if v, ok := fieldValue.(uint64); ok {
 				out.IsWritableBitmap = v
 			} else {
 				return out, errors.New("invalid type for IsWritableBitmap, expected uint64")
 			}
-		case "Accounts":
-			// Expect [][32]byte
-			if v, ok := fieldValue.([][32]byte); ok {
-				accounts := make([]solana.PublicKey, len(v))
-				for i, val := range v {
-					accounts[i] = solana.PublicKeyFromBytes(val[:])
-				}
-				out.Accounts = accounts
-			} else {
-				return out, errors.New("invalid type for Accounts, expected [][32]byte")
-			}
+			// we only need the keys to construct SVMExtraArgs, other keys can be skipped without return errors
```

Comment:
Hmm, I think the Encode() would need to also include the Accounts.
Atleast we need Accounts for MessageHasher.

Reply:
Yes, I added and mentioned in message hasher PR.